app/src/main/java/com/zeapo/pwdstore/sshkeygen/SshKeyGenFragment.kt

```diff
@@ -10,81 +10,128 @@ import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import android.view.inputmethod.InputMethodManager
-import android.widget.ArrayAdapter
-import android.widget.Button
-import android.widget.CheckBox
-import android.widget.EditText
-import android.widget.Spinner
+import android.widget.Toast
+import androidx.core.content.edit
 import androidx.core.content.getSystemService
 import androidx.fragment.app.Fragment
-import com.google.android.material.textfield.TextInputEditText
+import androidx.preference.PreferenceManager
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+import com.jcraft.jsch.JSch
+import com.jcraft.jsch.KeyPair
 import com.zeapo.pwdstore.R
+import com.zeapo.pwdstore.databinding.FragmentSshKeygenBinding
+import java.io.File
+import java.io.FileOutputStream
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
 
-class SshKeyGenFragment : Fragment() {
+class SshKeyGenFragment : Fragment(), CoroutineScope {
```

Comment:
Please don't extend the coroutine scope, create a new scope and cancel it onDestroyView. 

```
private val job = Job()
private val scope = CoroutineScope(job + Dispatchers.IO)
```

Also if we are using lifecycle components in v1 just do `lifecycleScope.launch`

Reply:
I second the use of `lifecycleScope`.